### PR TITLE
Allow wallet_addEthereumChain for ethereum chains

### LINKED ISSUE #13269: Allow wallet_addEthereumChain for Ethereum networks
With [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085), we're able to programmatically add new networks / chains to metamask which is a great experience for users versus having to manually add a custom RPC. 

However, [Metamask doesn't allow adding/updating chains programmatically](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js#L124) that have a chain id that corresponds to any of the default MetaMask chains (basically the Ethereum networks). Although adding a custom RPC will allow you to add an RPC with any of the default chain ids.

This would be really useful for allowing custom RPCs in a programmatic way or at the least be able to add duplicate Eth networks.

An example to validate:
```
const params = [{
    chainId: '0x1',
    chainName: 'Ethereum Mainnet (Localhost)',
    nativeCurrency: {
      name: 'Eth',
      symbol: 'Eth',
      decimals: 18
    },
    rpcUrls: ['https://localhost:8000'],
    blockExplorerUrls: ['https://etherscan.io/']
  }]

const args = { 
  method: 'wallet_addEthereumChain', 
  params: params
}
window.ethereum.request(args)
```

will result in 
```
MetaMask - RPC Error: May not specify default MetaMask chain. {code: -32602, message: 'May not specify default MetaMask chain.'}
```

_Note: I'm not able to create a feature request on community.metamask.io - there does not seem to be a way to post new topics._

### PR DESCRIPTION
**Pull request by kurushdubash**
Fixes: [#13629](https://github.com/MetaMask/metamask-extension/issues/13269)

Explanation:  Metamask allows adding a custom RPC with any of the default chain ids. However, it blocks requests for `wallet_addEthereumChain` when the `chain_id` is one of the defaults that come with Metamask. Removing this block would  allow `wallet_addEthereumChain` requests for Ethereum networks. This is a more convenient experience for users versus having to manually add a custom RPC and matches the functionality of manually adding an Ethereum chain.

Manual testing steps:  
  - ran `yarn tests` and verified all tests pass
  - ran `yarn build:test` and verified all tests pass
  - Manually sent a `wallet_addEthereumChain` request with an Ethereum `chain_id` and verified it was successful with no errors

### CODE REVIEW

**Comment by kurushdubash** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js [hunk 1]
```diff
@@ -121,14 +120,6 @@ async function addEthereumChainHandler(
...
   }
 
-  if (CHAIN_ID_TO_NETWORK_ID_MAP[_chainId]) {
-    return end(
-      ethErrors.rpc.invalidParams({
-        message: `May not specify default MetaMask chain.`,
-      }),
-    );
-  }
-
```
Comment (jpuri) [inline app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js:131]: I am not sure, id there any reason this check was put in place initially ?

Comment (kurushdubash) [reply app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js:131]: I could not find any documentation or information in commit history as to why that check is there. The same functionality is permitted through manual adding of RPC networks. In both flows, warnings are presented with regards to existing chain id networks.

**Comment by kurushdubash** [comment]
Hi @ryanml @jpuri , just wanted to bump this PR. There's been an increasing interest for allowing custom RPC endpoints for eth networks - https://twitter.com/MetaMask/status/1499448226180583429

**Comment by jordanmessina** [comment]
What is the status of this?

**Comment by brad-decker** [comment]
@adonesky1 can we possibly fix this PR up, merge it and then rebase your work in #16733  so that we incorporate this community  contribution.

**Comment by brad-decker** [comment]
@kurushdubash -- @adonesky1 cherry picked your commit into his branch so that you will get credit for your contribution. 🎉 please head over to #16733 for any futher comments on this thread.

**Comment by brad-decker** [comment]
@kurushdubash your commit landed as part of #16733 and you now have a shiny "contributor" label on your interactions with the repo:

image

Thanks @adonesky1 for taking the time to carry out my request to carry this commit into the codebase.

**Pull request closed without merging**